src/esp/gfx/Drawable.cpp

```diff
@@ -10,11 +10,25 @@
 
 namespace esp {
 namespace gfx {
-
+uint64_t Drawable::drawableIdCounter = 0;
 Drawable::Drawable(scene::SceneNode& node,
                    Magnum::GL::Mesh& mesh,
                    DrawableGroup* group /* = nullptr */)
-    : Magnum::SceneGraph::Drawable3D{node, group}, node_(node), mesh_(mesh) {}
+    : Magnum::SceneGraph::Drawable3D{node, group},
+      node_(node),
+      mesh_(mesh),
+      drawableId_(drawableIdCounter++) {
+  if (group) {
+    group->idToDrawable_.insert({drawableId_, this});
+  }
+}
+
+Drawable::~Drawable() {
+  DrawableGroup* group = drawables();
```

Comment:
I found a serious problem here.
In the dtor, It seems the usage of function `drawables()` has limitations. It cannot be used in this destructor.
The reason is that this group is deconstructed BEFORE this drawable is deconstructed. So when you call the `drawables()`, it would trigger the `CORRADE_ASSERT` within the `drawable()`

Thoughts?

Reply:
All right. I updated the code.
Current solution is:
When DrawableGroup get deconstructed earlier than the Drawables stored within, it will send a message to inform all the drawables that "no more query or operation on this group any more".